### ISSUE TITLE
Do not default to LTR for empty editors

### ISSUE THREAD

**Issue by mostafah** [facebook/draft-js#634]
**Do you want to request a *feature* or report a *bug*?**

Reporting a bug, I guess.

**What is the current behavior?**

On an empty Draft.js field, the first block has a `public-DraftStyleDefault-ltr` class and the caret is on the left side. This is before the first character is typed. This default does not make sense for an RTL environment, where everything else is RTL except for the caret.

**What is the expected behavior?**

Not setting a direction when no content is available. It would default to the parent direction, which is a sensible default.

**Which versions of Draft.js, and which browser / OS are affected by this issue? Did this work in previous versions of Draft.js?**

Draft 0.8.1, Chrome 52 and Safari 9.1.3, OS X 10.11.16. Didn’t check the older versions.

---

I’d be happy to work on a PR if you agree with the suggestion.

**Comment by stopachka** [comment]
Hey! This would be awesome. Feel free to pop in a PR @mostafah

**Comment by mostafah** [comment]
It seems like `fbjs` has [a default direction of LTR](https://github.com/facebook/fbjs/blob/e66ba20ad5be433eb54423f2b097d829324d9de6/packages/fbjs/src/unicode/UnicodeBidiDirection.js#L98). This default direction can not be neutral, must be either LTR or RTL.

I can fix it in my app by setting the global direction using `UnicodeBidiDirection.setGlobalDir(UnicodeBidiDirection.RTL)`, but since importing `fbjs` is not generally a good idea, I don’t think this is a proper solution.

What is the proper solution for this? Automatically setting global direction based on the parent or document direction, or exposing a function for setting the global direction?

**Comment by mostafah** [comment]
On a second thought, both of the above solutions have a global side effect, which is not good. What about having a `defaultDirection` prop on `Editor` component? That can then be sent as a `defaultDir` to `UnicodeBidiService`.

**Comment by mostafah** [comment]
Found a problem with that too!

It’s `EditorState` that creates and maintains direction map, not the `Editor` component. `EditorState` creates direction map in its `create` function. In other words, the initial direction map is calculated before the component is even created. So, having default direction as a prop does not work. (Unless we want to force a recalculation, but we don’t.)

So, it’s `EditorState` that can have a default direction property. Does that make sense to you?

**PS:** Sorry for the noise. I was not familiar with the internal design and code of Draft and am trying to find a proper design and implementation.

**Comment by sarmstrong** [comment]
We need to manually control RTL as well. Is that what you are ultimately looking for @mostafah? If so, I won't create a new issue. I think this would be a property that would be set on the EditorState during initial state creation. Then it could be used to override EditorBidiService.getDirectionMap if set.

**Comment by mostafah** [comment]
Yes, that’s what I’m after here. My original idea was to somehow remove the default direction, but after looking into the code I found out that that’s not possible without big changes to fbjs and Draft.js.

Since this would add a new property to `EditorState` I was waiting for a confirmation from the team, but I might as well go ahead and send a PR anyway.
